### PR TITLE
Hugo Docs Modification

### PR DESCRIPTION
**Pull request by GuessWhoSamFoo**
Branches off 389f9acd27032e7df8e63656277ff6c7d03e9052

To do:

- [x] Conversion script modifies /docs and /content with `show_in_list`
- [x] Find and replace /docs/assets with /content/assets/
- [ ] Under "Tools & Reference", move "SSH: Connect to Your Linode" files to `/networking/ssh`
- [ ] Rebase master and reapply script without losing existing changes
- [x] Check `/docs/contribute` works
- [x] Check for RSS feed references to /docs/index.xml
- [x] Create exception `/docs/contribute` for localhost test

Lower priority:
- [ ] Revisit ordering of guides in list
- [x] Update CI/test paths for new directory structure
- [x] Create decorator to replace all instances of `seek` in tests
- [x] Call scrapy from pytest to avoid csv intermediate
- [x] Hide search scores from users

Change notes (WIP):

 - Only one description within index files now. Markdown syntax applies within description. Add notes on single/double quote distinction.
 - Deprecated guides are now viewable when browsing by section
 - RSS link changed from `/docs/rss/index.xml` to `/docs/index.xml`; automatically updates based on publishing dates
 - `published` header is no longer optional.

### CODE REVIEW

File: docs/applications/configuration-management/beginners-guide-chef.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ The chef-client must be run with elevated privileges in order to properly config
 
 ### Run Lists
 
-Run lists define what cookbooks a node will use. The run list is an ordered list of all cookbooks and recipes that the chef-client needs to pull from the Chef server to run on a node. Run lists are also used to define [roles](http://docs.chef.io/server_manage_roles.html), which are used to define patterns and attributes across nodes.
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/beginners-guide-chef.md]: False positive

File: docs/applications/configuration-management/configure-and-use-salt-ssh.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ The Default location for the Roster file is: `/etc/salt/roster`.
     {: .note}
     >
     > Permissions leverage via sudo works only if the NOPASSWD option is set up for the user that is connecting to the minion in `/etc/sudoers`.
-    > More information on Roster files can be found in the [Roster files documentation](https://docs.saltstack.com/en/latest/topics/ssh/roster.html#ssh-roster).
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/configure-and-use-salt-ssh.md]: False positive

File: docs/applications/configuration-management/configure-and-use-salt-ssh.md [hunk 3]
```diff
@@ -164,7 +164,7 @@ The Default location for the Roster file is: `/etc/salt/roster`.
 
     {: .note}
     >
-    > A full list of execution modules is available at [Execution modules documentation](https://docs.saltstack.com/en/latest/ref/modules/all/index.html).
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/configure-and-use-salt-ssh.md]: Another one

File: docs/applications/configuration-management/install-and-configure-puppet.md [hunk 4]
```diff
@@ -13,12 +13,12 @@ published: 'Thursday, September 17th, 2015'
 title: Install and Configure Puppet
 external_resources:
     - '[Puppet Labs](https://puppetlabs.com/)'
-    - '[Puppet Open Source Documentation](https://docs.puppetlabs.com/puppet/)'
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/install-and-configure-puppet.md]: Flagged

File: docs/applications/configuration-management/install-and-configure-puppet.md [hunk 5]
```diff
@@ -62,7 +62,7 @@ Puppet can be used to manage multiple servers across various infrastructures, fr
     >
     >        wget https://apt.puppetlabs.com/puppetlabs-release-pc1-VERSION.deb
     >
-    >Any Ubuntu-specific commands will then have to be amended for the proper distribution. More information can be found in [Puppet's Installation Documentation](https://docs.puppetlabs.com/puppet/4.0/reference/install_linux.html#install-a-release-package-to-enable-puppet-labs-package-repositories) or our guide to [package management](https://www.linode.com/docs/tools-reference/linux-package-management).
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/install-and-configure-puppet.md]: Flagged

File: docs/applications/configuration-management/install-and-configure-puppet.md [hunk 6]
```diff
@@ -383,7 +383,7 @@ Although a new user has successfully been added to the Puppet master, the accoun
 
     {: .note}
     >
-    > The `file` directory is omitted from the `source` line because the `files` folder is the default location of files. For more information on the format used to access resources in a module, refer to the [official Puppet module documentation](https://docs.puppet.com/puppet/3.8/modules_fundamentals.html#module-layout).
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/install-and-configure-puppet.md]: Flagged

File: docs/applications/configuration-management/install-and-configure-puppet.md [hunk 7]
```diff
@@ -710,4 +710,4 @@ Now that the `accounts` and `firewall` modules have been created, tested, and ru
 
         sudo iptables -L
 
-Congratulations! You've successfully installed Puppet on a master and two agent nodes. Now that you've confirmed everything is working, you can create additional modules to automate configuration management on your agent nodes. For more information, see [Puppet module fundamentals](https://docs.puppet.com/puppet/latest/reference/modules_fundamentals.html). You can also install and use those modules others have created on the [Puppet Forge](https://docs.puppet.com/puppet/latest/reference/modules_installing.html).
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/install-and-configure-puppet.md]: Flagged

File: docs/applications/configuration-management/install-and-manage-mysql-databases-with-puppet-hiera-on-ubuntu-16-04.md [hunk 8]
```diff
@@ -15,40 +15,40 @@ contributor:
   link: https://tjll.net
 external_resources:
   - '[Puppet Labs](https://puppetlabs.com/)'
-  - '[Puppet Open Source Documentation](https://docs.puppetlabs.com/puppet/)'
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/install-and-manage-mysql-databases-with-puppet-hiera-on-ubuntu-16-04.md]: Flagged

File: docs/applications/configuration-management/install-and-manage-mysql-databases-with-puppet-hiera-on-ubuntu-16-04.md [hunk 9]
```diff
@@ -15,40 +15,40 @@ contributor:
...
+  - '[Puppet Open Source Documentation](https://content.puppetlabs.com/puppet/)'
   - '[The Puppet Forge](https://forge.puppet.com/)'
-  - '[Hiera documentation](https://docs.puppet.com/hiera/)'
-  - '[Facter](https://docs.puppet.com/facter/)'
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/install-and-manage-mysql-databases-with-puppet-hiera-on-ubuntu-16-04.md]: Flagged

File: docs/applications/configuration-management/install-and-manage-mysql-databases-with-puppet-hiera-on-ubuntu-16-04.md [hunk 10]
```diff
@@ -15,40 +15,40 @@ contributor:
...
+[Puppet](https://puppetlabs.com/) is a configuration management system that helps simplify the use and deployment of different types of software, making system administration more reliable and replicable. In this guide, we use Puppet to manage an installation of [MySQL](https://www.mysql.com/), a popular relational database used for applications such as WordPress, Ruby on Rails, and others. [Hiera](https://content.puppet.com/hiera/) is a method of defining configuration values that Puppet will use to simplify MySQL configuration.
 
-In this guide, you'll use Puppet to deploy [modules](https://docs.puppet.com/puppet/latest/modules_fundamentals.html) on your server. At the end, you will have MySQL installed, configured, and ready to use for a variety of applications that require a database backend.
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/install-and-manage-mysql-databases-with-puppet-hiera-on-ubuntu-16-04.md]: Flagged

File: docs/applications/configuration-management/learn-how-to-install-ansible-and-run-playbooks.md [hunk 11]
```diff
@@ -16,11 +16,11 @@ contributor:
     link: https://twitter.com/jlyman
 external_resources:
  - '[Ansible Home Page](http://www.ansible.com/home)'
- - '[Ansible Documentation](http://docs.ansible.com/ansible/index.html)'
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/learn-how-to-install-ansible-and-run-playbooks.md]: Flagged

File: docs/applications/configuration-management/learn-how-to-install-ansible-and-run-playbooks.md [hunk 12]
```diff
@@ -94,7 +94,7 @@ You were just able to get a valid connection to your server via Ansible!
 
 ## Using Ansible's Inventory File
 
-You executed an Ansible command against one client, but it would be cumbersome to have to type the host's address every single time, and what if you had several servers you wanted to apply the same configuration to? This is where Ansible's [inventory file](http://docs.ansible.com/ansible/intro_inventory.html) comes into play.
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/learn-how-to-install-ansible-and-run-playbooks.md]: Flagged

File: docs/applications/big-data/how-to-move-machine-learning-model-to-production.md [hunk 13]
```diff
@@ -74,73 +74,74 @@ Keras is a deep learning library for Python. It provides an object-oriented inte
...
         python ./mnist_model.py
 
     There may be a warning message as shown below from a pip or conda installation which means installing from source could offer superior performance.
 
-    {:.output}
-    ~~~
+    {{< output >}}
+~~~
     The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
     ~~~
```
Comment (GuessWhoSamFoo) [inline docs/applications/big-data/how-to-move-machine-learning-model-to-production.md]: Weird squiggly clean-up

File: docs/applications/cloud-storage/access-google-drive-linode.md [hunk 14]
```diff
@@ -3,26 +3,26 @@ author:
...
+modified: 2016-03-17
 modified_by:
     name: Linode
 title: 'Access Google Drive from Linode with Ubuntu 14.04'
 contributor:
     name: Scott Sumner
 ---
 
-*This is a Linode Community guide. [Write for us](/docs/contribute) and earn $250 per published guide.*
+*This is a Linode Community guide. [Write for us](/content/contribute) and earn $250 per published guide.*
```
Comment (GuessWhoSamFoo) [inline docs/applications/cloud-storage/access-google-drive-linode.md]: Check using /docs/contribute will actually work on Hugo. Or maybe just link to the full URL path.

File: docs/applications/configuration-management/use-salt-states-to-configure-a-lamp-stack-on-a-minion.md [hunk 15]
```diff
@@ -180,7 +180,7 @@ Salt State Modules are used for settings across groups of Minions. To adjust a c
         salt '<hostname or Minion ID>' cmd.run "a2ensite example.com.conf"
         salt '<hostname or Minion ID>' cmd.run "service apache2 reload"
 
-<a href="/docs/applications/salt/salt-states-apache-mysql-php-fail2ban" target="_blank">
+<a href="/content/applications/salt/salt-states-apache-mysql-php-fail2ban" target="_blank">
```
Comment (GuessWhoSamFoo) [inline docs/applications/configuration-management/use-salt-states-to-configure-a-lamp-stack-on-a-minion.md]: Check for HTML tags like this. Not sure why they're used.

File: docs/applications/containers/how-to-install-docker-and-deploy-a-lamp-stack.md [hunk 16]
```diff
@@ -87,11 +87,16 @@ When an image downloads, there are no image containers running.
...
-    >
-    >The website's root directory is `/var/www/example.com/public_html/`.
+    {{< note >}}
+The website's root directory is `/var/www/example.com/public_html/`.
+{{< /note >}}
 
 Congratulations, you have installed a configured LAMP stack using Docker!
 
-##Where to Find Configuration Settings
+<<<<<<< HEAD
```
Comment (GuessWhoSamFoo) [inline docs/applications/containers/how-to-install-docker-and-deploy-a-lamp-stack.md]: Unresolved merge conflict. Grep for `<<<<<< HEAD`

File: docs/applications/messaging/installing-prosody-xmpp-server-on-ubuntu-10-04-lucid.md [hunk 17]
```diff
@@ -4,32 +4,32 @@ author:
...
 
-Before we begin with the installation and configuration of Prosody, we assume that you have a running and up to date installation of Ubuntu 10.04 (Lucid), have completed our [getting started](/docs/getting-started/) guide, and have logged in via SSH as root.
+Before we begin with the installation and configuration of Prosody, we assume that you have a running and up to date installation of Ubuntu 10.04 (Lucid), have completed our [getting started](/content/getting-started/) guide, and have logged in via SSH as root.
 
 Adding Software Repositories
 ----------------------------
```
Comment (GuessWhoSamFoo) [inline docs/applications/messaging/installing-prosody-xmpp-server-on-ubuntu-10-04-lucid.md]: This style formatting makes it difficult to know which header is being used. Deprecate in favor of `#, ##, ###` etc

File: docs/applications/big-data/how-to-move-machine-learning-model-to-production.md (hunk 13 above)
Comment (GuessWhoSamFoo) [reply docs/applications/big-data/how-to-move-machine-learning-model-to-production.md]: @EdwardAngert Can you fix?

File: docs/applications/configuration-management/use-salt-states-to-configure-a-lamp-stack-on-a-minion.md (hunk 15 above)
Comment (GuessWhoSamFoo) [reply docs/applications/configuration-management/use-salt-states-to-configure-a-lamp-stack-on-a-minion.md]: @cwlinode There may be cases where is is needed, but can you find all cases of `<a href=''>` tags and see if they can be replaced?

File: docs/applications/messaging/installing-prosody-xmpp-server-on-ubuntu-10-04-lucid.md (hunk 17 above)
Comment (GuessWhoSamFoo) [reply docs/applications/messaging/installing-prosody-xmpp-server-on-ubuntu-10-04-lucid.md]: @Guaris Can you fix?

Comment (Guaris) [reply docs/applications/messaging/installing-prosody-xmpp-server-on-ubuntu-10-04-lucid.md]: no prob

File: docs/applications/cloud-storage/access-google-drive-linode.md (hunk 14 above)
Comment (GuessWhoSamFoo) [reply docs/applications/cloud-storage/access-google-drive-linode.md]: @Jskobos Can you look into this?

**Comment by GuessWhoSamFoo** [comment]
[Hugo: fix errors from live migration](https://trello.com/c/yvzUSRG6/19-hugo-fix-errors-from-live-migration)

File: docs/applications/configuration-management/use-salt-states-to-configure-a-lamp-stack-on-a-minion.md (hunk 15 above)
Comment (cwlinode) [reply docs/applications/configuration-management/use-salt-states-to-configure-a-lamp-stack-on-a-minion.md]: @GuessWhoSamFoo All set: https://github.com/linode/docs/pull/1138

File: docs/applications/big-data/how-to-move-machine-learning-model-to-production.md (hunk 13 above)
Comment (EdwardAngert) [reply docs/applications/big-data/how-to-move-machine-learning-model-to-production.md]: find squiggles, remove squiggles, realign

```
    {{< output >}}
stuff to output aligns left
then the closing /output does too
{{< /output>
```

File: docs/web-servers/apache/running-fastcgi-php-fpm-on-debian-7-with-apache.md [hunk 18]
```diff
@@ -11,7 +11,7 @@ contributor:
     link: https://twitter.com/jesin_a
 external_resources:
  - '[PHP-FPM configuration directives](http://php.net/manual/en/install.fpm.configuration.php)'
- - '[mod_fastcgi configuration directives](http://www.fastcgi.com/mod_fastcgi/docs/mod_fastcgi.html)'
+ - '[mod_fastcgi configuration directives](http://www.fastcgi.com/mod_fastcgi/content/mod_fastcgi.html)'
```
Comment (GuessWhoSamFoo) [inline docs/web-servers/apache/running-fastcgi-php-fpm-on-debian-7-with-apache.md]: Change back to docs

File: docs/security/upgrading/updating-virtual-host-settings-from-apache-2-2-to-apache-2-4.md [hunk 19]
```diff
@@ -116,7 +116,7 @@ If you are utilizing access control rules within your virtual host files, you wi
         systemctl reload httpd.service
 
     {{< note >}}
-For more information on how you can enable or restrict access to your websites with various `Require` lines, see the [Apache website](http://httpd.apache.org/docs/current/howto/access.html). Most users will want to use the `Require all granted` line, but there may be exceptions.
+For more information on how you can enable or restrict access to your websites with various `Require` lines, see the [Apache website](http://httpd.apache.org/content/current/howto/access.html). Most users will want to use the `Require all granted` line, but there may be exceptions.
```
Comment (GuessWhoSamFoo) [inline docs/security/upgrading/updating-virtual-host-settings-from-apache-2-2-to-apache-2-4.md]: Change back to docs

File: docs/databases/hadoop/install-configure-run-spark-on-top-of-hadoop-yarn-cluster.md [hunk 20]
```diff
@@ -270,12 +270,12 @@ The Spark shell provides an interactive way to examine and work with your data.
...
-The Scala Spark API is beyond the scope of this guide. You can find the official documentation on [Official Apache Spark documentation](https://spark.apache.org/docs/latest/quick-start.html).
+The Scala Spark API is beyond the scope of this guide. You can find the official documentation on [Official Apache Spark documentation](https://spark.apache.org/content/latest/quick-start.html).
 
 ## Where to Go Next ?
 
 Now that you have a running Spark cluster, you can:
 
-- Learn any of the Scala, Java, Python, or R APIs to create Spark applications from the [Apache Spark Programming Guide](https://spark.apache.org/docs/latest/rdd-programming-guide.html)
-- Interact with your data with [Spark SQL](https://spark.apache.org/docs/latest/sql-programming-guide.html)
+- Learn any of the Scala, Java, Python, or R APIs to create Spark applications from the [Apache Spark Programming Guide](https://spark.apache.org/content/latest/rdd-programming-guide.html)
```
Comment (GuessWhoSamFoo) [inline docs/databases/hadoop/install-configure-run-spark-on-top-of-hadoop-yarn-cluster.md]: Back to docs

File: docs/databases/hadoop/install-configure-run-spark-on-top-of-hadoop-yarn-cluster.md [hunk 21]
```diff
@@ -270,12 +270,12 @@ The Spark shell provides an interactive way to examine and work with your data.
         // Count the number of non blank lines
         input.filter(line => line.length()>0).count()
 
-The Scala Spark API is beyond the scope of this guide. You can find the official documentation on [Official Apache Spark documentation](https://spark.apache.org/docs/latest/quick-start.html).
+The Scala Spark API is beyond the scope of this guide. You can find the official documentation on [Official Apache Spark documentation](https://spark.apache.org/content/latest/quick-start.html).
```
Comment (GuessWhoSamFoo) [inline docs/databases/hadoop/install-configure-run-spark-on-top-of-hadoop-yarn-cluster.md]: Back to docs

File: docs/assets/httpd.conf [hunk 22]
```diff
@@ -1,9 +1,9 @@
 #
 # This is the main Apache HTTP server configuration file.  It contains the
 # configuration directives that give the server its instructions.
-# See <URL:http://httpd.apache.org/docs/2.4/> for detailed information.
+# See <URL:http://httpd.apache.org/content/2.4/> for detailed information.
```
Comment (GuessWhoSamFoo) [inline docs/assets/httpd.conf]: Change 3 instances back to docs

File: docs/assets/apache2.conf [hunk 23]
```diff
@@ -1,6 +1,6 @@
 # This is the main Apache server configuration file.  It contains the
 # configuration directives that give the server its instructions.
-# See http://httpd.apache.org/docs/2.4/ for detailed information about
+# See http://httpd.apache.org/content/2.4/ for detailed information about
```
Comment (GuessWhoSamFoo) [inline docs/assets/apache2.conf]: Same

File: docs/applications/configuration-management/install-and-manage-mysql-databases-with-puppet-hiera-on-ubuntu-16-04.md [hunk 24]
```diff
@@ -118,18 +119,16 @@ The following configuration will define Puppet variables in `common.yaml` to inj
...
 Hiera configuration files are formatted as yaml, with keys defining the Puppet parameters to inject their associated values. To get started,  set the MySQL root password. The following example of a Puppet manifest is one way to control this password:
 
-{: .file}
-:   ~~~ pp
-    class { '::mysql::server':
-      root_password => 'examplepassword',
-    }
-    ~~~
-
+{{< file ":   ~~~ pp" >}}
```
Comment (EdwardAngert) [inline docs/applications/configuration-management/install-and-manage-mysql-databases-with-puppet-hiera-on-ubuntu-16-04.md]: this whole section needs a proper looking at
saving for later @GuessWhoSamFoo

**Pull request merged**